lib/mapper.js

```diff
@@ -72,13 +72,21 @@ exports.track = function(track) {
       }
     });
   } else {
+    if (typeof track.proxy('context.traits.email') !== 'undefined'){
```

Comment:
thanks chris! we really appreciate your contribution :)

i _think_ the reason to proxy directly from `context.traits.email` was to avoid first looking at `properties.email` for this particular case (since as an *event property*, `email` could be an indirect object grammatically speaking) and *only* respect `context.traits.email` (which describes the email of the subject who took the action).

agree with prateek's take on the local variable approach from a style perspective :)